### PR TITLE
bpo-41171: Added method "PyMetaType_FromSpec" to allow for specifying the metaclass for the type.

### PR DESCRIPTION
**Pull request by WildCard65**
https://bugs.python.org/issue41171

### CODE REVIEW

**Comment by WildCard65** [comment]
Paging @vstinner for review.

File: Objects/typeobject.c [hunk 1]
```diff
@@ -2880,14 +2880,19 @@ static const short slotoffsets[] = {
...
-PyType_FromSpecWithBases(PyType_Spec *spec, PyObject *bases)
+PyMetaType_FromSpecWithBases(PyTypeObject *meta_type, PyType_Spec *spec, PyObject *bases)
 {
-    return PyType_FromModuleAndSpec(NULL, spec, bases);
+    return PyMetaType_FromModuleAndSpec(NULL, meta_type, spec, bases);
 }
 
 PyObject *
-PyType_FromModuleAndSpec(PyObject *module, PyType_Spec *spec, PyObject *bases)
+PyMetaType_FromModuleAndSpec(PyObject *module, PyTypeObject *meta_type, PyType_Spec *spec, PyObject *bases)
```
Comment (vstinner) [inline Objects/typeobject.c]: I propose the name PyMetaType_FromSpec and remove PyMetaType_FromSpecWithBases.

Comment (WildCard65) [reply Objects/typeobject.c]: So, you're suggesting renaming PyMetaType_FromSpecWithBases to just PyMetaType_FromSpec or just keeping the PyMetaType_FromSpec from above?

Comment (vstinner) [reply Objects/typeobject.c]: Rename PyMetaType_FromModuleAndSpec (4 parameters) to PyMetaType_FromSpec. And so remove the other PyMetaType_FromSpec (2 parameters).

Comment (WildCard65) [reply Objects/typeobject.c]: Should I move the method out of "Py_LIMITED_API" then?

**Comment by WildCard65** [comment]
@vstinner I performed the change you requested.

**Pull request closed without merging**